daemon/cmd/daemon_main.go

```diff
@@ -1803,15 +1803,16 @@ func initKubeProxyReplacementOptions() {
 	} else {
 		if option.Config.Tunnel != option.TunnelDisabled &&
 			option.Config.NodePortMode != option.NodePortModeSNAT {
-
 			log.Warnf("Disabling NodePort's %q mode feature due to tunneling mode being enabled",
 				option.Config.NodePortMode)
 			option.Config.NodePortMode = option.NodePortModeSNAT
 		}
 
-		option.Config.NodePortHairpin =
-			option.Config.NodePortAcceleration != option.NodePortAccelerationDisabled ||
-				len(option.Config.Devices) == 1
+		option.Config.NodePortHairpin = len(option.Config.Devices) == 1
+		if option.Config.NodePortAcceleration != option.NodePortAccelerationDisabled &&
+			len(option.Config.Devices) != 1 {
+			log.Fatalf("Cannot set NodePort acceleration due to multi-device setup (%q). Specify --%s with a single device to enable NodePort acceleration.", option.Config.Devices, option.Devices)
+		}
```

Comment:
@brb PTAL in context of the autodetection logic and how the devices list may be pre-populated here.